src/services/ApiService.ts

```diff
@@ -522,14 +523,38 @@ const getAvailableConnections = async (
 }
 
 const getTransactionHistory = async (
-  address: string
+  walletAnalyticsRequest: WalletAnalyticsRequest
 ): Promise<WalletAnalytics> => {
   const config = ConfigService.getInstance().getConfig()
 
-  const response = await request<WalletAnalytics>(
-    `${config.apiUrl}/analytics/wallets/${address}`
+  if (!walletAnalyticsRequest.fromTimestamp) {
+    throw new ValidationError('Required parameter "fromTimestamp" is missing.')
+  }
+
+  if (!walletAnalyticsRequest.toTimestamp) {
+    throw new ValidationError('Required parameter "toTimestamp" is missing.')
+  }
+
+  const url = new URL(
+    `${config.apiUrl}/analytics/wallets/${walletAnalyticsRequest.walletAddress}`
+  )
+
+  url.searchParams.append(
+    'fromTimestamp',
+    walletAnalyticsRequest.fromTimestamp.toString()
   )
 
+  url.searchParams.append(
+    'toTimestamp',
+    walletAnalyticsRequest.toTimestamp.toString()
+  )
+
+  if (config.integrator) {
+    url.searchParams.append('integrator', config.integrator)
+  }
```

Comment:
We add this automatically inside request function 🙂

Reply:
Inside the request method, we are passing it in headers and here the backend as a field in query params.

Seeing that, I additionally added the integrator as a query param here. If it's the same and backend always picks up from the headers then I'll go ahead and remove this.

Reply:
Upon discussion, resorting to passing integrator via headers in request function and migrating to query param in v3 probably